walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,67 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> options = new ArrayList<>();
+    List<Event> eventsSorted = new ArrayList<>();
+    eventsSorted.addAll(events);
+    Collections.sort(
+        eventsSorted,
+        new Comparator<Event>() {
+          @Override
+          public int compare(Event a, Event b) {
+            return TimeRange.ORDER_BY_START.compare(a.getWhen(), b.getWhen());
+          }
+        });
+    eventsSorted.removeIf(
+        new Predicate<Event>() {
+          @Override
+          public boolean test(Event e) {
+            Collection<String> eventAttendees = new ArrayList<>();
+            eventAttendees.addAll(e.getAttendees());
+            eventAttendees.removeAll(request.getAttendees());
+            return e.getAttendees().size() == eventAttendees.size();
+          }
+        });
```

Comment:
i think you should put this code into its own method/function inside the class and then call it — this is just a suggestion, but it would make your code more readable to know what you are doing in each of these large code blocks. it seems like you do this at the end with ```addTimeRangeIfPossible```

Reply:
Ah that's a good point, I'll refactor that out so that it's easier to follow.